src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheck.java

```diff
@@ -293,9 +298,19 @@ private boolean isEmptyLoopBodyAllowed(DetailAST ast) {
      */
     private static boolean hasUnbracedStatements(DetailAST ast) {
         final DetailAST nextSibling = ast.getNextSibling();
-        return nextSibling != null
+        final DetailAST parent = ast.getParent();
+        boolean result = false;
+
+        if (parent.getType() == TokenTypes.SWITCH_RULE) {
+            final DetailAST lastSibling = getLastSibling(ast);
```

Comment:
`getLastSibling` == `parent.getLastChild()`, no reason for new method here.

Reply:
Done.